instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcTest.java

```diff
@@ -56,6 +59,83 @@ protected InstrumentationExtension testing() {
     return testing;
   }
 
+  @Test
+  void grpcAttributesExtractor() throws Exception {
```

Comment:
```suggestion
  void captureRequestMetadata() throws Exception {
```

Also, can we move this test case to the `AbstractGrpcTest` and make sure that both javaagent and library instrumentations' tests cover it?

Reply:
@mateuszrzeszutek How would you go about doing this?
AbstractGrpcTest has no access to setCapturedRequestMetadata because it's in another module

Reply:
You can take a look at e.g. Armeria tests for some inspiration.
For example, the library instrumentation module sets the captured request headers in test code: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/6f9ca0b80bb732100fca157e989f9b0da3c93f5a/instrumentation/armeria-1.3/library/src/test/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaHttpServerTest.java#L23-L27
The javaagent simply passes it as a configuration property -- it's done generically on the test agent level for HTTP tests and you'll probably want to set this config just for the grpc instrumentation, so here's another example of setting custom properties in the build files: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/6f9ca0b80bb732100fca157e989f9b0da3c93f5a/instrumentation/reactor/reactor-netty/reactor-netty-1.0/javaagent/build.gradle.kts#L49-L50

Reply:
@mateuszrzeszutek Thanks for the advice! it worked out great
you really make this first contribution experience much more approachable